File: app/src/main/java/ethiopia/covid/android/ui/fragment/ContentState.java
package ethiopia.covid.android.ui.fragment;

import android.view.View;

import ethiopia.covid.android.R;

/**
 * Created by BrookMG on 4/10/2020 in ethiopia.covid.android.ui.fragment
 * inside the project CoVidEt .
 */
public class ContentState {

    static void changeProgressBarVisibility(View workingFrom, boolean show) {
        workingFrom.findViewById(R.id.main_progress_bar).setVisibility( show ? View.VISIBLE : View.GONE);
    }

    static void changeErrorDialogVisibility(View workingFrom, boolean show) {
        workingFrom.findViewById(R.id.error_indicator).setVisibility( show ? View.VISIBLE : View.GONE);
        workingFrom.findViewById(R.id.refresh_button).setVisibility( show ? View.VISIBLE : View.GONE);
        workingFrom.findViewById(R.id.error_text).setVisibility( show ? View.VISIBLE : View.GONE);
    }

    static void setRefreshButtonAction(View workingFrom , View.OnClickListener listener) {
        workingFrom.findViewById(R.id.refresh_button).setOnClickListener(listener);
    }

}

File: app/src/main/java/ethiopia/covid/android/ui/fragment/ContentState.kt
package ethiopia.covid.android.ui.fragment

import android.view.View
import ethiopia.covid.android.R
import ethiopia.covid.android.databinding.ContentStateLayoutBinding

/**
 * Created by BrookMG on 4/10/2020 in ethiopia.covid.android.ui.fragment
 * inside the project CoVidEt .
 */
object ContentState {
    @JvmStatic
    fun changeProgressBarVisibility(workingFrom: ContentStateLayoutBinding, show: Boolean) {
        workingFrom.mainProgressBar.visibility = if (show) View.VISIBLE else View.GONE
    }

    @JvmStatic
    fun changeErrorDialogVisibility(workingFrom: ContentStateLayoutBinding, show: Boolean) {
        workingFrom.errorIndicator.visibility = if (show) View.VISIBLE else View.GONE
        workingFrom.refreshButton.visibility = if (show) View.VISIBLE else View.GONE
        workingFrom.errorText.visibility = if (show) View.VISIBLE else View.GONE
    }

    @JvmStatic
    fun setRefreshButtonAction(workingFrom: ContentStateLayoutBinding, listener: View.OnClickListener?) {
        workingFrom.refreshButton.setOnClickListener(listener)
    }
}